src/www/widgets/widgets/system_log.widget.php

```diff
@@ -28,67 +29,146 @@
 
 require_once("guiconfig.inc");
 
-$system_logfile = '/var/log/system.log';
 
-if (!$config['widgets']['systemlogfiltercount']){
-  $syslogEntriesToFetch = 20;
+if (!$config['widgets']['systemlogfiltercount']) {
+    $systemlogEntriesToFetch = 20;
 } else {
-  $syslogEntriesToFetch = $config['widgets']['systemlogfiltercount'];
+    $systemlogEntriesToFetch = $config['widgets']['systemlogfiltercount'];
 }
+$input_errors = array();
 
-if (is_numeric($_POST['logfiltercount'])) {
-   $countReceived =  $_POST['logfiltercount'];
-   $config['widgets']['systemlogfiltercount'] = $countReceived;
-   write_config("Saved Widget System Log Filter Setting");
-   header(url_safe('Location: /index.php'));
-   exit;
+if (isset($_COOKIE['inputerrors'])) {
```

Comment:
I'm not a fan of this. Broken filter string == no output. While the idea is nice this is probably overly complicating the rework and review.

Reply:
Hi! Sorry, I didn't quite understand. Don't like the idea of using a filter? or implementing a check and error return via cookies?

Reply:
Hi :) Second part specifically, yes. I'm unsure about the first part but I think it can be integrated if we discuss further.

Reply:
just didn't want to write an arbitrary line to the config, and it will still be sanitized by api when the query is executed (and the filtering result may be unexpected). so trying to inform the user that the filter string does not meet the requirements.
couldn't find a better way to return an error for the widget.  maybe there is a better way i can try?

Reply:
ok, how about this: ideally I'd like to merge this but without the filter and we dicuss filter inclusion in next PR?

Reply:
sure ) let's try. I will prepare a request without a filter. thanks!